lib/matplotlib/tests/test_collections.py

```diff
@@ -830,6 +831,56 @@ def test_collection_set_verts_array():
         assert np.array_equal(ap._codes, atp._codes)
 
 
+def _fig_to_bytes(fig):
+    with io.BytesIO() as bio:
+        fig.savefig(bio)
+        return bio.getvalue()
+
+
+@pytest.fixture()
+def ind():
+    return np.linspace(16, 365, (365-16)*4)
+
+
+@pytest.fixture()
+def dep1(ind):
+    return np.sin(2*np.pi*ind/153) + np.cos(2*np.pi*ind/127)
+
+
+@pytest.fixture()
+def dep2(dep1):
+    return dep1 + .2
+
+
+@pytest.mark.parametrize("fname", ["fill_between", "fill_betweenx"])
+def test_fbp_collection_set_data(fname, ind, dep1, dep2):
```

Comment:
I'm quite strongly against using fixtures for test data, because it reduces readability through indirection and introduces coupling between test.

This is much clearer: 
```suggestion
def test_fbp_collection_set_data(fname):
    ind = np.linspace(16, 365, (365-16)*4)
    dep1 = np.sin(2*np.pi*ind/153) + np.cos(2*np.pi*ind/127)
    dep2 = dep1 + 0.2
```

Note that you can simplify the test data (and here also using one of the suggested new naming conventions:
```suggestion
def test_fbp_collection_set_data(fname):
    t = np.linspace(0, 10)
    f1 = np.sin(t)
    f2 = f1 + 0.2
```

Reply:
38cd7b9214c1813281f931b72e572b1b5edd2cf7